core/store/benches/finalize_bench.rs

```diff
@@ -0,0 +1,271 @@
+//! Benchmarks for the in-memory preparation of writes before the commit, aka
+//! `finalize`.
+//!
+//! All writes are batched to a DB transaction before being committed. This
+//! involves copying around the data and deserializing it.
+//!
+//! Since this is done on the hot path, whereas the actual commit can often be
+//! done in a different thread, we have this benchmark to test the speed.
+//! 
+//! Right now, the module only contains test for `ShardChunk` and
+//! `PartialEncodedChunk`. These have been observed to become quite large and
+//! have ab impact on the overall client performance.
+
+#[macro_use]
+extern crate bencher;
+
+use bencher::{black_box, Bencher};
+use borsh::BorshSerialize;
+use near_chain::Chain;
+use near_chunks::ShardsManager;
+use near_crypto::{InMemorySigner, KeyType, Signer};
+use near_primitives::hash::CryptoHash;
+use near_primitives::merkle::{merklize, MerklePathItem};
+use near_primitives::receipt::{ActionReceipt, DataReceipt, Receipt, ReceiptEnum};
+use near_primitives::shard_layout::ShardLayout;
+use near_primitives::sharding::{
+    ChunkHash, EncodedShardChunk, PartialEncodedChunk, PartialEncodedChunkPart,
+    PartialEncodedChunkV2, ReceiptProof, ReedSolomonWrapper, ShardChunk, ShardChunkHeader,
+    ShardChunkHeaderV3, ShardChunkV2, ShardProof,
+};
+use near_primitives::transaction::{Action, FunctionCallAction, SignedTransaction};
+use near_primitives::types::AccountId;
+use near_primitives::validator_signer::InMemoryValidatorSigner;
+use near_store::DBCol;
+use rand::prelude::SliceRandom;
+
+/// `ShardChunk` -> `StoreUpdate::insert_ser`.
+///
+/// ~10ms or faster for ~24MB receipts
+fn benchmark_write_shard_chunk(bench: &mut Bencher) {
+    let transactions = vec![];
+    let receipts = create_benchmark_receipts();
+    let chunk_hash: ChunkHash = CryptoHash::default().into();
+    let chunk = create_shard_chunk(&chunk_hash, transactions, receipts);
+
+    let chunks = spread_in_memory(chunk);
+
+    let store = near_store::test_utils::create_test_store();
+    bench.iter(|| {
```

Comment:
Just for my understanding, is the following statement correct?
```This benchmark measures the time it takes to copy data from a random place in memory to the store update. ```

Reply:
Yes, it correct but I would extend it to:
This benchmark measures the time it takes to *serialize* and copy data from a random place in memory to the store_update.

Also, "copy" and "serialization" are implementation details that we might want to change. Ideally, we would not be copying anything at this point. So it's conceptually better to say, this measures the time to record a serialized insertion in the store_update.